independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/InstanceImpl.java

```diff
@@ -120,7 +113,7 @@ public boolean isAmbiguous() {
     }
 
     @Override
-    public void destroy(T instance) {
+    public void destroy(Object instance) {
```

Comment:
What `Object`, the instance should still be of type `T`, right?

Reply:
Because of the generics hell... yes, it should be `T` but it should be safe use `Object`.

Reply:
Yea, `Object` works too. It's just that I saw `T` everywhere else, so this struck me as weird :)